src/content/docs/paper/dev/api/dialogs.mdx

```diff
@@ -0,0 +1,366 @@
+        Key key = event.getIdentifier();
+
+        if (key.equals(Key.key("papermc:paperchan/disagree"))) {
+            // If the identifier is the same as the disagree one, set the connection result to false
+            setConnectionJoinResult(event.getCommonConnection(), false);
+        } else if (key.equals(Key.key("papermc:paperchan/agree"))) {
+            // If it is the same as the agree one, set the result to true
+            setConnectionJoinResult(event.getCommonConnection(), true);
+        }
+    }
+
+    /**
+     * Simple utility method for setting a connection's dialog response result.
+     */
+    private void setConnectionJoinResult(PlayerCommonConnection connection, boolean value) {
+        CompletableFuture<Boolean> future = awaitingResponse.get(connection);
+        if (future != null) {
+            future.complete(value);
+        }
+    }
+}
+```
+
+And that's all there is to it. You can use this code to block players from joining your server before they should
+be allowed to.
+
+<details>
+  <summary>In-game preview</summary>
+  <Video src={DialogShowcaseMp4} />
+</details>
+
+## Example: Retrieving and parsing user input
+The dialog for this example will be fairly simple: It again confirmation-type dialog which contains two number range inputs.
+The top input will be for setting the level, the bottom input for setting the experience percentage towards the next level.
+When the player clicks on the confirmation button, they should have their levels and exp set to the configured values.
+
+```java
+Dialog.create(builder -> builder.empty()
+    .base(DialogBase.builder(Component.text("Configure your new experience value"))
+        .inputs(List.of(
+            DialogInput.numberRange("level", Component.text("Level", NamedTextColor.GREEN), 0f, 100f)
+                .step(1f)
+                .initial(0f)
+                .width(300)
+                .build(),
+            DialogInput.numberRange("experience", Component.text("Experience", NamedTextColor.GREEN), 0f, 100f)
+                .step(1f)
+                .initial(0f)
+                .labelFormat("%s: %s percent to the next level")
+                .width(300)
+                .build()
+        ))
+        .build()
+    )
+    .type(DialogType.confirmation(
+        ActionButton.create(
+            Component.text("Confirm", TextColor.color(0xAEFFC1)),
+            Component.text("Click to confirm your input."),
+            100,
+            DialogAction.customClick(Key.key("papermc:user_input/confirm"), null)
+        ),
+        ActionButton.create(
+            Component.text("Discard", TextColor.color(0xFFA0B1)),
+            Component.text("Click to discard your input."),
+            100,
+            null // If we set the action to null, it doesn't do anything and closes the dialog
+        )
+    ))
+);
+```
+
+### Reading the input
+To retrieve the values the user put into the dialog, we once again have to listen to a `PlayerCustomClickEvent`.
```

Comment:
What about custom click action with the callback built in, instead of listening to the event?